src/System.Net.Http/src/System/Net/Http/Headers/HeaderDescriptor.cs

```diff
@@ -99,19 +102,58 @@ public string GetHeaderValue(ReadOnlySpan<byte> headerValue)
             }
 
             // If it's a known header value, use the known value instead of allocating a new string.
-            if (_knownHeader != null && _knownHeader.KnownValues != null)
+            if (_knownHeader != null)
             {
-                string[] knownValues = _knownHeader.KnownValues;
-                for (int i = 0; i < knownValues.Length; i++)
+                if (_knownHeader.KnownValues != null)
+                {
+                    string[] knownValues = _knownHeader.KnownValues;
+                    for (int i = 0; i < knownValues.Length; i++)
+                    {
+                        if (ByteArrayHelpers.EqualsOrdinalAsciiIgnoreCase(knownValues[i], headerValue))
+                        {
+                            return knownValues[i];
+                        }
+                    }
+                }
+
+                if (KnownHeader == KnownHeaders.Location)
                 {
-                    if (ByteArrayHelpers.EqualsOrdinalAsciiIgnoreCase(knownValues[i], headerValue))
+                    // Normally Location should be in ISO-8859-1, ocassionally some servers respond with UTF-8 though
+                    if (TryDecodeUtf8(headerValue, out string decoded))
                     {
-                        return knownValues[i];
+                        return decoded;
                     }
                 }
             }
 
             return HttpRuleParser.DefaultHttpEncoding.GetString(headerValue);
         }
+
+        private static bool TryDecodeUtf8(ReadOnlySpan<byte> input, out string decoded)
+        {
+            // TODO: Utilize vectorization helpers here if/when they're made public
```

Comment:
Is there an issue number we can reference here with this TODO?  Or perhaps at least more detail in this comment about where this "vectorization helpers" are?

Reply:
I'm actually waiting for @GrabYourPitchforks to come back from vacations before I merge this - wanted to talk with him before I create one